jet_tracking/datastream.py

```diff
@@ -829,108 +1153,487 @@ def editor(self, im):
+        critical_prob = 1 - alpha/2
+        degrees_of_freedom = len(com) - 2
+        tinv = lambda p, df: abs(stats.t.ppf(p/2., df))
+        ts = tinv(alpha, degrees_of_freedom)
+        y = np.append(y, [y_min, y_max])
+        if slope:
+            x_model = (y - intercept)*(1/slope)
+            x_model_plus = (y - intercept - ts*intercept_err)*(1 / (slope + ts*slope_err))
+            x_model_minus = (y - intercept + ts * intercept_err) * (1 / (slope - ts * slope_err))
+            yl = list(y)
+            i_max = yl.index(max(yl))
+            i_min = yl.index(min(yl))
+            self.context.best_fit_line = [[int(yl[i_min]), int(x_model[i_min])],
+                                  [int(yl[i_max]), int(x_model[i_max])]]
+            self.context.best_fit_line_plus = [[int(yl[i_min]), int(x_model_plus[i_min])],
+                                       [int(yl[i_max]), int(x_model_plus[i_max])]]
+            self.context.best_fit_line_minus = [[int(yl[i_min]), int(x_model_minus[i_min])],
+                                        [int(yl[i_max]), int(x_model_minus[i_max])]]
+            return True
+        else:
+            return False
+
+    def run_image_thread(self):
+        """
+        Run the image processing thread.
+
+        This function processes the image if the thread is not paused and the camera is connected.
+        If live data is available, it retrieves the image from the camera. Otherwise, it generates
+        a simulated image. The image is then processed using the editor function. If the
+        find_com_bool flag is set, the center of the jet is located in the image. Finally, the
+        processed image is emitted through the camImager signal.
+        """
+        if not self.paused:
             if self.connected:
-                image = caget(self.cam_name + ':IMAGE2:ArrayData')
-                image = self.fix_image(image, self.array_size_x_viewer,
-                                       self.array_size_y_viewer)
+                if self.context.live_data:
+                    image_array = caget(self.cam_name + ':IMAGE1:ArrayData')
+                    image = np.reshape(image_array, (self.array_size_y_viewer,
+                                           self.array_size_x_viewer))
+                else:
+                    self.cam_name.gen_image()
+                    image = self.cam_name.jet_im
                 image = self.editor(image)
-                qimage = array2qimage(image)
-                self.signals.camImage.emit(qimage)
-                time.sleep(1/self.refresh_rate)
-            else:
-                print("you are not connected")
-                time.sleep(1/self.refresh_rate)
-        print("Interruption was requested: Image Thread")
+                if self.find_com_bool:
+                    self.find_center(image)
+                self.signals.camImager.emit(image)
+        time.sleep(1/self.refresh_rate)
+
+
+class MotorThread(QObject):
```

Comment:
This was the way it was done before (overriding the run method of QThread). However, after a bit of research I found that the "movetothread" method is the accepted method in the community. Now, if you ask me why, I can't give you a detailed answer but there are a lot of explanations out there if you do a search for it. The main reason I found I needed to change it was because all signals were being received and processed on the main thread instead of the respective threads.